ci_framework/roles/libvirt_manager/tasks/attach_interface.yml

```diff
@@ -58,20 +58,24 @@
         mac_address: "{{ _lm_mac_address }}"
       ansible.builtin.import_tasks: reserve_ip.yml
 
+    - name: Ensure network name is valid
+      when: network.name | length > 15
+      ansible.builtin.fail:
+        msg: "Invalid network name. Length greater than the allowed limit."
```

Comment:
why not truncate directly?

Reply:
I agree. 

Looking back a bit, I believe there would be a failure in `ci-nmstate` role as it wouldn't be able to apply/create the connection.

Do you believe, we would still need to truncate?

Reply:
probably yes, for cases that aren't passing via ci_nmstate :).

Reply:
Agreed. will make the modifications in the next commit.